src/temporal_terms.jl

```diff
@@ -0,0 +1,56 @@
+# Basic Lag:
+
+# syntax:
+"""
+    lag(term, nsteps)
+This `@formula` term is used to introduce lagged variables.
+For example `lag(x,1)` effectively adds a new column containing
+the value of the `x` column from the previous row.
+If there are no such row (e.g. because this is the first row), then the lagged column will
+contain `missing`.
+
+Note: this is only a basic row-wise lag operation.
+It is up to the user to ensure that that there data is sorted by the temporal variable,
+and that their observations are spaced with regular time-steps.
+(Which may require adding extra-rows filled with `missing` values.)
+"""
+lag(term, nsteps) = error("`lag`  should only be used within a @formula");
+
+# type of model where syntax applies:
+const LAG_CONTEXT = Any
```

Comment:
I don't see the point of this. Why not use `Any` directly?

Reply:
This is basically copied from the docs on extending.
I think it has some merit, but i can remove it if you think that makes sense to

Reply:
Well, that's useful in the docs, but since it's `Any` in this case it doesn't add anything IMO.

Reply:
Yeah in the docs it's meant to be pedagogical.